core/src/main/java/org/jboss/pnc/rex/core/RemoteEntityClient.java

```diff
@@ -187,4 +198,45 @@ private org.jboss.pnc.api.dto.Request getCallbackRequest(String taskName, String
         }
         return callbackRequest;
     }
+
+    /**
+     * get the task results of the direct dependencies of the task, only if the configuration allows so
+     *
+     * @param task task to return task results
+     * @return Map of taskName and the result if configuration allows sharing of results, otherwise null
+     */
+    private Map<String, Object> getTaskResults(Task task) {
```

Comment:
I like this method in general and I think it could be useful. Maybe what if we moved the data loading part into TaskRegistry.

Reply:
Will do!